plugins/inputs/procstat/procstat.go

```diff
@@ -519,8 +640,46 @@ func (p *Procstat) Init() error {
 	return nil
 }
 
+func containsIP(a []net.IP, x net.IP) bool {
+	for _, n := range a {
+		if x.Equal(n) {
+			return true
+		}
+	}
+	return false
+}
+
+func isIPV4(ip net.IP) bool {
+	return ip.To4() != nil
+}
+
+func isIPV6(ip net.IP) bool {
+	return ip.To4() == nil
+}
+
+// endpointString return the correct representation of ip and port for IPv4 and IPv6
+func endpointString(ip net.IP, port uint32) string {
+	if isIPV6(ip) {
+		return fmt.Sprintf("[%s]:%d", ip, port)
+	}
+	return fmt.Sprintf("%s:%d", ip, port)
+}
+
+// metricEnabled check is some group of metrics are enabled in the config file
+func (p *Procstat) metricEnabled(m string) bool {
+	for _, n := range p.MetricsInclude {
+		if m == n {
+			return true
+		}
+	}
+	return false
+}
+
```

Comment:
Please use the [already existing function](https://github.com/influxdata/telegraf/blob/3bcf72293d23b9730b7f0d2559d91d6468370ed8/internal/choice/choice.go#L8).